pkg/policy/execute.go

```diff
@@ -65,12 +69,45 @@ type Executor struct {
 
 // QueryResult contains the result information from an executed query.
 type QueryResult struct {
-	Name        string          `json:"name"`
-	Description string          `json:"description"`
-	Columns     []string        `json:"result_headers"`
-	Data        [][]interface{} `json:"result_rows"`
-	Type        QueryType       `json:"type"`
-	Passed      bool            `json:"check_passed"`
+	Name         string          `json:"name"`
+	Description  string          `json:"description"`
+	QueryColumns []string        `json:"-"`
+	Columns      []string        `json:"result_header"`
+	Data         [][]interface{} `json:"-"`
+	Rows         Rows            `json:"result_rows"`
+	Type         QueryType       `json:"type"`
+	Passed       bool            `json:"check_passed"`
+}
+
+type Row struct {
+	// AdditionalData is any extra information that was returned from the result set
+	AdditionalData []interface{} `json:"additional_data,omitempty"`
+	// Identifiers are a list of identifiers as defined by the policy
+	Identifiers []interface{} `json:"identifiers,omitempty"`
+	// Reason is a user readable explanation returned by the query, or interpolated from check defined reason.
+	Reason string `json:"reason,omitempty"`
+	// Status is user defined status of the row i.e OK, ALERT etc'
+	Status string `json:"status,omitempty"`
+}
+type Rows []Row
+
+func (r Rows) Len() int {
+	return len(r)
+}
+
+func (r Rows) Swap(i, j int) {
+	r[i], r[j] = r[j], r[i]
+}
+
+func (r Rows) Less(i, j int) bool {
+	r1 := r[i]
+	r2 := r[j]
+	for l := 0; l < len(r1.Identifiers); l++ {
+		if cast.ToString(r1.Identifiers[l]) > cast.ToString(r2.Identifiers[l]) {
```

Comment:
Is this descending sort?

Reply:
ill add a test to this, but does it matter? the sort is mostly to keep it consistent with same data

Reply:
test would be good yes